src/groupeddataframe/groupeddataframe.jl

```diff
@@ -1126,20 +1127,50 @@ First Group (1 row): g = 1
      │ Int64  Char
 ─────┼─────────────
    1 │     1  a
+
+julia> filter(:x => x -> x[1] == 'a', gd, ungroup=true)
+1×2 DataFrame
+ Row │ g      x
+     │ Int64  Char
+─────┼─────────────
+   1 │     1  a
 ```
 """
-Base.filter(f, gdf::GroupedDataFrame) =
-    gdf[[f(sdf)::Bool for sdf in gdf]]
-Base.filter((col, f)::Pair{<:ColumnIndex}, gdf::GroupedDataFrame) =
-    _filter_helper(gdf, f, gdf.idx, gdf.starts, gdf.ends, parent(gdf)[!, col])
-Base.filter((cols, f)::Pair{<:AbstractVector{Symbol}}, gdf::GroupedDataFrame) =
-    filter([index(parent(gdf))[col] for col in cols] => f, gdf)
-Base.filter((cols, f)::Pair{<:AbstractVector{<:AbstractString}}, gdf::GroupedDataFrame) =
-    filter([index(parent(gdf))[col] for col in cols] => f, gdf)
-Base.filter((cols, f)::Pair, gdf::GroupedDataFrame) =
-    filter(index(parent(gdf))[cols] => f, gdf)
-Base.filter((cols, f)::Pair{<:AbstractVector{Int}}, gdf::GroupedDataFrame) =
-    _filter_helper(gdf, f, gdf.idx, gdf.starts, gdf.ends, (parent(gdf)[!, i] for i in cols)...)
+function Base.filter(f, gdf::GroupedDataFrame; ungroup::Bool=false)
+    res = gdf[[f(sdf)::Bool for sdf in gdf]]
+    return ungroup ? DataFrame(res) : res
+end
```

Comment:
IIRC in other places we used `@inline` to ensure the return type is inferred? Here it could make sense to do something like:
```julia
@inline function Base.filter(gdf::Any, gdf::GroupedDataFrame; ungroup::Bool=false)
    res = _filter(f, gdf)
    return ungroup ? DataFrame(res) : res
end
```
And handle the different kinds of arguments in `_filter`?

Reply:
OK